src/api/app/controllers/source_controller.rb

```diff
@@ -516,13 +524,14 @@ def update_project_meta
     render_ok
   end
 
-  def check_and_remove_repositories!(repositories, full_remove, force = false)
-    error = Project.check_repositories(repositories) unless force
-    if !force && error[:error]
+  def check_and_remove_repositories!(repositories, opts)
+    error = Project.check_repositories(repositories) unless opts[:force]
+    if !opts[:force] && error[:error]
       raise RepoDependency, error[:error]
     else
-      error = Project.remove_repositories(repositories, full_remove)
-      if !force && error[:error]
+      error = Project.remove_repositories(repositories, opts)
```

Comment:
we can do so, but please with another pull request. Keep in mind this must be backported and we should have as few changes as needed for that.